src/views/Collectibles/hooks/useBunnySpecialLottery.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState, useCallback } from 'react'
+import { useAppDispatch } from 'state'
+import { useWeb3React } from '@web3-react/core'
+import { useGetCurrentLotteryId } from 'state/lottery/hooks'
+import { getGraphLotteryUser } from 'state/lottery/getUserLotteryData'
+import { UserRound } from 'state/types'
+import { getBunnySpecialLotteryAddress } from 'utils/addressHelpers'
+import { multicallv2 } from 'utils/multicall'
+import bunnySpecialLotteryAbi from 'config/abi/bunnySpecialLottery.json'
+import Nfts from 'config/constants/nfts'
+import { fetchCurrentLotteryId } from 'state/lottery'
+import { getBallerClaim, getLottieClaim, getLuckyClaim, NftClaim, NO_CLAIM } from '../helpers'
+
+const useBunnySpecialLottery = (): {
+  canClaimLottie: () => Promise<NftClaim>
+  canClaimLucky: () => Promise<NftClaim>
+  canClaimBaller: () => Promise<NftClaim>
+} => {
+  const { account } = useWeb3React()
+  const dispatch = useAppDispatch()
+  const currentLotteryId = useGetCurrentLotteryId()
+  const currentLotteryIdNum = currentLotteryId && parseInt(currentLotteryId)
+  const [userRounds, setUserRounds] = useState<UserRound[]>(null)
+
+  useEffect(() => {
+    dispatch(fetchCurrentLotteryId())
+  }, [dispatch])
+
+  useEffect(() => {
+    setUserRounds(null)
+  }, [account])
+
+  useEffect(() => {
+    const getUserData = async () => {
+      const lotteryNftContractAddress = getBunnySpecialLotteryAddress()
+      const calls = [
+        {
+          name: 'startLotteryRound',
+          address: lotteryNftContractAddress,
+          params: [],
+        },
+        {
+          name: 'finalLotteryRound',
+          address: lotteryNftContractAddress,
+          params: [],
+        },
+      ]
+
+      const [[startRound], [finalRound]] = await multicallv2(bunnySpecialLotteryAbi, calls)
+      const startLotteryRoundInt = startRound.toNumber()
+      const finalLotteryRoundInt = finalRound.toNumber()
+
+      const lotteryIdsArray = []
+      for (let i = startLotteryRoundInt; i <= finalLotteryRoundInt; i++) {
+        lotteryIdsArray.push(`${i}`)
+      }
+      const userParticipation = await getGraphLotteryUser(account, undefined, undefined, {
+        lottery_in: lotteryIdsArray,
+      })
+      setUserRounds(userParticipation.rounds)
+    }
+
+    if (account && currentLotteryIdNum && !userRounds) {
+      getUserData()
+    }
+  }, [currentLotteryIdNum, userRounds, account])
```

Comment:
Before this was two separate effects - however one was dependent on the other so I combined them for readability.